### DISCUSSION TITLE
Longhorn engine path mismatch

### DISCUSSION THREAD

**Discussion by rwlove** [longhorn/longhorn#3782] [Q&A] [answered]
I recently tore-down and rebuilt my cluster and I'm hitting the following error.

I'm using Longhorn 1.2.4.

2022/03/23 22:25:55 proto: duplicate proto type registered: VersionResponse
time="2022-03-23T22:25:55Z" level=info msg="Start overwriting built-in settings with customized values"
W0323 22:25:55.381618       1 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2022-03-23T22:25:55Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/*], output , stderr, ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/*': No such file or directory\n, error exit status 2"
time="2022-03-23T22:25:55Z" level=info msg="Skip the leader election for the upgrade since the current Longhorn system is already up to date"
W0323 22:25:55.452817       1 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0323 22:25:55.473476       1 shared_informer.go:223] Waiting for caches to sync for longhorn datastore
I0323 22:25:56.673719       1 shared_informer.go:230] Caches are synced for longhorn datastore 
W0323 22:25:56.673900       1 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.

and ultimately

time="2022-03-23T22:26:00Z" level=debug msg="CheckEngineImageReadiness: nodes [worker4.foo.com] don't have the engine image longhornio/longhorn-engine:v1.2.4"

I think the problem is that my engine-binaries/ is in /var/lib/longhorn/ and not /var/lib/rancher/longhorn/ and I'm not sure where this is configured.

➜  ~ ls /var/lib/longhorn/engine-binaries -l
total 0
drwxr-xr-x 2 root root  6 Mar 23 11:51 longhornio-longhorn-engine-v1.1.2
drwxr-xr-x 2 root root 22 Mar 23 18:25 longhornio-longhorn-engine-v1.2.4

but the longhorn-manager is looking for them in /var/lib/rancher and I'm not sure why.

I'm installing longhorn on a vanilla kubernetes 1.23.5 cluster via the helm chart.

Any ideas where my misconfiguration is or if this is a bug?

**Comment by PhanLe1010** [comment, marked as answer]
> time="2022-03-23T22:25:55Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/], output , stderr, ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/': No such file or directory\n, error exit status 2"

This is not an issue. The log level is `level=info`
